src/hooks/useIsVisible.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useRef, useState } from 'react'
+
+/** Determines if an element is partially visible in the viewport. */
+const useIsVisible = <T extends HTMLElement>(initialValue = false): [boolean, React.RefObject<T>] => {
+  const [isVisible, setIsVisible] = useState(initialValue)
+  const elementRef = useRef<T | null>(null)
+
+  /** Determines if the element is visible in the viewport. */
+  const checkVisibility = () => {
+    const element = elementRef.current
+
+    if (element) {
+      const rect = element.getBoundingClientRect()
+      const top = rect.top < window.innerHeight
+      const bottom = rect.bottom > 0
+      const left = rect.left < window.innerWidth
+      const right = rect.right > 0
+      const visible = top && bottom && left && right
+      setIsVisible(visible)
+    }
+  }
+
+  useEffect(() => {
+    checkVisibility()
+    window.addEventListener('scroll', checkVisibility)
```

Comment:
This can definitely use the throttled scrollTop from scrollTopStore without any performance impact (see other comment)

Reply:
Yep! Switched here also: 5c38140d2a2e3dbe314612bfb23fd87fe8474dcb